src/ArborX_LinearBVH.hpp

```diff
@@ -39,20 +39,21 @@ template <typename BVH>
 struct DistributedTreeNearestUtils;
 } // namespace Details
 
-template <typename MemorySpace, typename Enable = void>
-class BoundingVolumeHierarchy
+template <typename MemorySpace, typename BoundingVolume = Box,
+          typename Enable = void>
+class BasicBoundingVolumeHierarchy
```

Comment:
I don't like the name `BasicBoundingVolumeHierarchy`. There's nothing basic about it. What's wrong with `BoundingVolumeHierarchyBase`?

Reply:
I find `BoundingVolumeHierarchyBase` potentially misleading as it usually denotes inheritance relationship.
`BasicBoundingVolumeHierarchy` follows the naming of the standard strings library (`std::string` is an alias for `std::basic_string<char>`).  Similarly the `midspan` proposal has a `basic_mdspan`.

Reply:
I guess I would be ok now that you introduced more cases of such words. Personally, I still don't understand the meaning behind that word here due to insufficient English. I looked Merriam-Webster's [basic](https://www.merriam-webster.com/dictionary/basic) page. In what sense is it being used here,
- 1a : of, relating to, or forming the base or essence, or
- 2 : constituting or serving as the basis or starting point?
I don't know if any of this fits. I must be missing something.